### PR TITLE
SDK-152 [Java SDK] add default regional auth servers

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-152](https://incountry.atlassian.net/browse/SDK-152)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: src/test/java/com/incountry/residence/sdk/HttpDaoImplTests.java [hunk 1]
```diff
@@ -334,11 +334,28 @@ void testSearchPopApiResponse() throws StorageClientException, StorageServerExce
...
+                countryLoadBadResponseNullName,
+                countryLoadBadResponseEmptyName,
+                countryLoadBadResponseNullId,
+                countryLoadBadResponseEmptyId,
+                countryLoadBadResponseEmptyCountries));
         Dao dao = new HttpDaoImpl(null, null, null, agent);
         assertNotNull(dao);
         StorageServerException ex = assertThrows(StorageServerException.class, () -> new HttpDaoImpl(null, null, null, agent));
         assertEquals("Response error", ex.getMessage());
+        assertEquals("java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $", ex.getCause().getMessage());
```
Comment (artemgolozubov) [inline src/test/java/com/incountry/residence/sdk/HttpDaoImplTests.java:348]: This line looks suspicious. AFAIK our SDKs should throw their own exceptions (like StorageServerException), and should not pass low-level exceptions to user.

Comment (VladimirLevitskiy) [reply src/test/java/com/incountry/residence/sdk/HttpDaoImplTests.java:348]: Our SDK throws StorageServerException in this case, IllegalStateException is incapsulated into it

File: src/main/java/com/incountry/residence/sdk/tools/dao/impl/HttpDaoImpl.java [hunk 2]
```diff
@@ -79,7 +80,7 @@ public HttpDaoImpl(String endPoint, String endpointMask, String countriesEndpoin
     private String initUsingDefaultEndpoint(boolean defaultEndpoint, String mask) {
         String resultMask = null;
         if (defaultEndpoint) {
-            resultMask = "." + (mask != null ? mask : DEFAULT_ENDPOINT_MASK);
+            resultMask = (mask != null ? mask : DEFAULT_ENDPOINT_MASK);
```
Comment (artemgolozubov) [inline src/main/java/com/incountry/residence/sdk/tools/dao/impl/HttpDaoImpl.java:83]: Do we still need parentheses here? Also it seems that this line could be simplified:
`resultMask = mask || DEFAULT_ENDPOINT_MASK;`

File: src/main/java/com/incountry/residence/sdk/tools/dao/impl/HttpDaoImpl.java [hunk 3]
```diff
@@ -112,19 +113,19 @@ private EndPoint getEndpoint(String country) throws StorageServerException {
...
 
     private EndPoint getDefaultEndpointForPop(POP pop, String country) {
         EndPoint resultEndpoint;
         if (pop != null) { //mid pop for default endpoint
-            resultEndpoint = new EndPoint(pop.getHost(), pop.getHost());
+            resultEndpoint = new EndPoint(pop.getHost(), pop.getHost(), pop.getRegion(DEFAULT_REGION));
         } else { //minipop for default endpoint
             if (endPointMask == null) {
-                resultEndpoint = new EndPoint(endPoint, getAudienceForMiniPop(endPoint, country));
+                resultEndpoint = new EndPoint(endPointUrl, getAudienceForMiniPop(endPointUrl, country), DEFAULT_REGION);
```
[the remark this answers is not in the document]
Comment (VladimirLevitskiy) [reply src/main/java/com/incountry/residence/sdk/tools/dao/impl/HttpDaoImpl.java]: We need this condition for case, when we use default endpoint with custom endpoint mask, and they are not suit for each other

File: src/main/java/com/incountry/residence/sdk/tools/dao/impl/HttpDaoImpl.java [hunk 4]
```diff
@@ -103,24 +103,26 @@ private void loadCountries() throws StorageServerException {
...
-                pop = popMap.get(country);
-            }
+            POP pop = getPopIfCountryIsMidPop(country);
             if (pop != null) { //mid pop for default endpoint
                 return new EndPoint(pop.getHost(), pop.getHost(), pop.getRegion(DEFAULT_REGION));
-            } else if (endPointMask != null) {
-                String mainUrl = URI_HTTPS + DEFAULT_COUNTRY + endPointMask;
-                return new EndPoint(mainUrl, getAudienceForMiniPop(mainUrl, country), DEFAULT_REGION);
             }
+            String mainUrl = URI_HTTPS + DEFAULT_COUNTRY + usingDefaultEndpointMask;
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/residence/sdk/tools/dao/impl/HttpDaoImpl.java:110]: usingDefaultEndpointMask is kinda boolean-ish name. And anyways it's hard to guess var's purpose. Please, rename :)

**Pull request merged**